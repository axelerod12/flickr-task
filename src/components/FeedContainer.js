import React from 'react';
import Masonry from 'react-masonry-component';
import Feed from './Feed';

export default function FeedContainer(props) {

  return (
    <div className="container">
      <Masonry>
        {
          props.flickrFeed.map((feed, index) => {
            return <Feed key={index} data={feed} />
          })
        }
      </Masonry>
    </div>
  )
}