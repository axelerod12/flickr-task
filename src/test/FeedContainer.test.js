import React from 'react';
import { shallow } from 'enzyme';
import FeedContainer from '../components/FeedContainer';

const props = [{ "id": "40575384753", "owner": "59464817@N02", "secret": "310af33489", "server": "7828", "farm": 8, "title": "Svassi Maggiori", "ispublic": 1, "isfriend": 0, "isfamily": 0, "description": { "_content": "Podiceps cristatus\nGreat Crested Grebe\nPeschiera del Garda" }, "ownername": "Fausto Deseri", "tags": "greatcrestedgrebe podicepscristatus svassomaggiore corteggiamento peschieradelgarda wildlife nature birds wildanimals nikond500 afsnikkor300mmf28difedii nikontc17eii faustodeseri", "url_s": "https:\/\/live.staticflickr.com\/7828\/40575384753_310af33489_m.jpg", "height_s": "160", "width_s": "240" }, { "id": "47541388921", "owner": "115045266@N08", "secret": "213f95c299", "server": "7866", "farm": 8, "title": "street g\u00e9om\u00e9trique", "ispublic": 1, "isfriend": 0, "isfamily": 0, "description": { "_content": "" }, "ownername": "Rudy Pilarski", "tags": "structure street nikon nb bw b\u00e2timent building blackandwhite noiretblanc monochrome perspective paris people personne geometry g\u00e9om\u00e9trie g\u00e9om\u00e9tria g\u00e9om\u00e9trique graphique graphic urbain urban urbano urbanisme france francia europe europa jussieux architecture architectura architectural dowtown thebestoffnikon thepassionphotography d7100 design", "url_s": "https:\/\/live.staticflickr.com\/7866\/47541388921_213f95c299_m.jpg", "height_s": "240", "width_s": "160" }, { "id": "46817753384", "owner": "163510631@N05", "secret": "3a8d43ea24", "server": "7926", "farm": 8, "title": "birds , birds , birds", "ispublic": 1, "isfriend": 0, "isfamily": 0, "description": { "_content": "OLYMPUS DIGITAL CAMERA" }, "ownername": "evablanchardcouet", "tags": "nature birds tree outside", "url_s": "https:\/\/live.staticflickr.com\/7926\/46817753384_3a8d43ea24_m.jpg", "height_s": "156", "width_s": "240" }];

describe('FeedContainer Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FeedContainer flickrFeed={props} />);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <MasonryComponent />', () => {
    expect(wrapper.find('MasonryComponent')).toHaveLength(1);
  });
});