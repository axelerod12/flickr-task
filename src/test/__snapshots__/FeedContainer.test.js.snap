// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedContainer Component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FeedContainer
    flickrFeed={
      Array [
        Object {
          "description": Object {
            "_content": "Podiceps cristatus
Great Crested Grebe
Peschiera del Garda",
          },
          "farm": 8,
          "height_s": "160",
          "id": "40575384753",
          "isfamily": 0,
          "isfriend": 0,
          "ispublic": 1,
          "owner": "59464817@N02",
          "ownername": "Fausto Deseri",
          "secret": "310af33489",
          "server": "7828",
          "tags": "greatcrestedgrebe podicepscristatus svassomaggiore corteggiamento peschieradelgarda wildlife nature birds wildanimals nikond500 afsnikkor300mmf28difedii nikontc17eii faustodeseri",
          "title": "Svassi Maggiori",
          "url_s": "https://live.staticflickr.com/7828/40575384753_310af33489_m.jpg",
          "width_s": "240",
        },
        Object {
          "description": Object {
            "_content": "",
          },
          "farm": 8,
          "height_s": "240",
          "id": "47541388921",
          "isfamily": 0,
          "isfriend": 0,
          "ispublic": 1,
          "owner": "115045266@N08",
          "ownername": "Rudy Pilarski",
          "secret": "213f95c299",
          "server": "7866",
          "tags": "structure street nikon nb bw bâtiment building blackandwhite noiretblanc monochrome perspective paris people personne geometry géométrie géométria géométrique graphique graphic urbain urban urbano urbanisme france francia europe europa jussieux architecture architectura architectural dowtown thebestoffnikon thepassionphotography d7100 design",
          "title": "street géométrique",
          "url_s": "https://live.staticflickr.com/7866/47541388921_213f95c299_m.jpg",
          "width_s": "160",
        },
        Object {
          "description": Object {
            "_content": "OLYMPUS DIGITAL CAMERA",
          },
          "farm": 8,
          "height_s": "156",
          "id": "46817753384",
          "isfamily": 0,
          "isfriend": 0,
          "ispublic": 1,
          "owner": "163510631@N05",
          "ownername": "evablanchardcouet",
          "secret": "3a8d43ea24",
          "server": "7926",
          "tags": "nature birds tree outside",
          "title": "birds , birds , birds",
          "url_s": "https://live.staticflickr.com/7926/46817753384_3a8d43ea24_m.jpg",
          "width_s": "240",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <MasonryComponent
        className=""
        disableImagesLoaded={false}
        elementType="div"
        enableResizableChildren={false}
        imagesLoadedOptions={Object {}}
        onLayoutComplete={[Function]}
        onRemoveComplete={[Function]}
        options={Object {}}
        updateOnEachImageLoad={false}
      >
        <Feed
          data={
            Object {
              "description": Object {
                "_content": "Podiceps cristatus
Great Crested Grebe
Peschiera del Garda",
              },
              "farm": 8,
              "height_s": "160",
              "id": "40575384753",
              "isfamily": 0,
              "isfriend": 0,
              "ispublic": 1,
              "owner": "59464817@N02",
              "ownername": "Fausto Deseri",
              "secret": "310af33489",
              "server": "7828",
              "tags": "greatcrestedgrebe podicepscristatus svassomaggiore corteggiamento peschieradelgarda wildlife nature birds wildanimals nikond500 afsnikkor300mmf28difedii nikontc17eii faustodeseri",
              "title": "Svassi Maggiori",
              "url_s": "https://live.staticflickr.com/7828/40575384753_310af33489_m.jpg",
              "width_s": "240",
            }
          }
        />
        <Feed
          data={
            Object {
              "description": Object {
                "_content": "",
              },
              "farm": 8,
              "height_s": "240",
              "id": "47541388921",
              "isfamily": 0,
              "isfriend": 0,
              "ispublic": 1,
              "owner": "115045266@N08",
              "ownername": "Rudy Pilarski",
              "secret": "213f95c299",
              "server": "7866",
              "tags": "structure street nikon nb bw bâtiment building blackandwhite noiretblanc monochrome perspective paris people personne geometry géométrie géométria géométrique graphique graphic urbain urban urbano urbanisme france francia europe europa jussieux architecture architectura architectural dowtown thebestoffnikon thepassionphotography d7100 design",
              "title": "street géométrique",
              "url_s": "https://live.staticflickr.com/7866/47541388921_213f95c299_m.jpg",
              "width_s": "160",
            }
          }
        />
        <Feed
          data={
            Object {
              "description": Object {
                "_content": "OLYMPUS DIGITAL CAMERA",
              },
              "farm": 8,
              "height_s": "156",
              "id": "46817753384",
              "isfamily": 0,
              "isfriend": 0,
              "ispublic": 1,
              "owner": "163510631@N05",
              "ownername": "evablanchardcouet",
              "secret": "3a8d43ea24",
              "server": "7926",
              "tags": "nature birds tree outside",
              "title": "birds , birds , birds",
              "url_s": "https://live.staticflickr.com/7926/46817753384_3a8d43ea24_m.jpg",
              "width_s": "240",
            }
          }
        />
      </MasonryComponent>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Feed
            data={
              Object {
                "description": Object {
                  "_content": "Podiceps cristatus
Great Crested Grebe
Peschiera del Garda",
                },
                "farm": 8,
                "height_s": "160",
                "id": "40575384753",
                "isfamily": 0,
                "isfriend": 0,
                "ispublic": 1,
                "owner": "59464817@N02",
                "ownername": "Fausto Deseri",
                "secret": "310af33489",
                "server": "7828",
                "tags": "greatcrestedgrebe podicepscristatus svassomaggiore corteggiamento peschieradelgarda wildlife nature birds wildanimals nikond500 afsnikkor300mmf28difedii nikontc17eii faustodeseri",
                "title": "Svassi Maggiori",
                "url_s": "https://live.staticflickr.com/7828/40575384753_310af33489_m.jpg",
                "width_s": "240",
              }
            }
          />,
          <Feed
            data={
              Object {
                "description": Object {
                  "_content": "",
                },
                "farm": 8,
                "height_s": "240",
                "id": "47541388921",
                "isfamily": 0,
                "isfriend": 0,
                "ispublic": 1,
                "owner": "115045266@N08",
                "ownername": "Rudy Pilarski",
                "secret": "213f95c299",
                "server": "7866",
                "tags": "structure street nikon nb bw bâtiment building blackandwhite noiretblanc monochrome perspective paris people personne geometry géométrie géométria géométrique graphique graphic urbain urban urbano urbanisme france francia europe europa jussieux architecture architectura architectural dowtown thebestoffnikon thepassionphotography d7100 design",
                "title": "street géométrique",
                "url_s": "https://live.staticflickr.com/7866/47541388921_213f95c299_m.jpg",
                "width_s": "160",
              }
            }
          />,
          <Feed
            data={
              Object {
                "description": Object {
                  "_content": "OLYMPUS DIGITAL CAMERA",
                },
                "farm": 8,
                "height_s": "156",
                "id": "46817753384",
                "isfamily": 0,
                "isfriend": 0,
                "ispublic": 1,
                "owner": "163510631@N05",
                "ownername": "evablanchardcouet",
                "secret": "3a8d43ea24",
                "server": "7926",
                "tags": "nature birds tree outside",
                "title": "birds , birds , birds",
                "url_s": "https://live.staticflickr.com/7926/46817753384_3a8d43ea24_m.jpg",
                "width_s": "240",
              }
            }
          />,
        ],
        "className": "",
        "disableImagesLoaded": false,
        "elementType": "div",
        "enableResizableChildren": false,
        "imagesLoadedOptions": Object {},
        "onLayoutComplete": [Function],
        "onRemoveComplete": [Function],
        "options": Object {},
        "updateOnEachImageLoad": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "description": Object {
                "_content": "Podiceps cristatus
Great Crested Grebe
Peschiera del Garda",
              },
              "farm": 8,
              "height_s": "160",
              "id": "40575384753",
              "isfamily": 0,
              "isfriend": 0,
              "ispublic": 1,
              "owner": "59464817@N02",
              "ownername": "Fausto Deseri",
              "secret": "310af33489",
              "server": "7828",
              "tags": "greatcrestedgrebe podicepscristatus svassomaggiore corteggiamento peschieradelgarda wildlife nature birds wildanimals nikond500 afsnikkor300mmf28difedii nikontc17eii faustodeseri",
              "title": "Svassi Maggiori",
              "url_s": "https://live.staticflickr.com/7828/40575384753_310af33489_m.jpg",
              "width_s": "240",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "description": Object {
                "_content": "",
              },
              "farm": 8,
              "height_s": "240",
              "id": "47541388921",
              "isfamily": 0,
              "isfriend": 0,
              "ispublic": 1,
              "owner": "115045266@N08",
              "ownername": "Rudy Pilarski",
              "secret": "213f95c299",
              "server": "7866",
              "tags": "structure street nikon nb bw bâtiment building blackandwhite noiretblanc monochrome perspective paris people personne geometry géométrie géométria géométrique graphique graphic urbain urban urbano urbanisme france francia europe europa jussieux architecture architectura architectural dowtown thebestoffnikon thepassionphotography d7100 design",
              "title": "street géométrique",
              "url_s": "https://live.staticflickr.com/7866/47541388921_213f95c299_m.jpg",
              "width_s": "160",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "description": Object {
                "_content": "OLYMPUS DIGITAL CAMERA",
              },
              "farm": 8,
              "height_s": "156",
              "id": "46817753384",
              "isfamily": 0,
              "isfriend": 0,
              "ispublic": 1,
              "owner": "163510631@N05",
              "ownername": "evablanchardcouet",
              "secret": "3a8d43ea24",
              "server": "7926",
              "tags": "nature birds tree outside",
              "title": "birds , birds , birds",
              "url_s": "https://live.staticflickr.com/7926/46817753384_3a8d43ea24_m.jpg",
              "width_s": "240",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MasonryComponent
          className=""
          disableImagesLoaded={false}
          elementType="div"
          enableResizableChildren={false}
          imagesLoadedOptions={Object {}}
          onLayoutComplete={[Function]}
          onRemoveComplete={[Function]}
          options={Object {}}
          updateOnEachImageLoad={false}
        >
          <Feed
            data={
              Object {
                "description": Object {
                  "_content": "Podiceps cristatus
Great Crested Grebe
Peschiera del Garda",
                },
                "farm": 8,
                "height_s": "160",
                "id": "40575384753",
                "isfamily": 0,
                "isfriend": 0,
                "ispublic": 1,
                "owner": "59464817@N02",
                "ownername": "Fausto Deseri",
                "secret": "310af33489",
                "server": "7828",
                "tags": "greatcrestedgrebe podicepscristatus svassomaggiore corteggiamento peschieradelgarda wildlife nature birds wildanimals nikond500 afsnikkor300mmf28difedii nikontc17eii faustodeseri",
                "title": "Svassi Maggiori",
                "url_s": "https://live.staticflickr.com/7828/40575384753_310af33489_m.jpg",
                "width_s": "240",
              }
            }
          />
          <Feed
            data={
              Object {
                "description": Object {
                  "_content": "",
                },
                "farm": 8,
                "height_s": "240",
                "id": "47541388921",
                "isfamily": 0,
                "isfriend": 0,
                "ispublic": 1,
                "owner": "115045266@N08",
                "ownername": "Rudy Pilarski",
                "secret": "213f95c299",
                "server": "7866",
                "tags": "structure street nikon nb bw bâtiment building blackandwhite noiretblanc monochrome perspective paris people personne geometry géométrie géométria géométrique graphique graphic urbain urban urbano urbanisme france francia europe europa jussieux architecture architectura architectural dowtown thebestoffnikon thepassionphotography d7100 design",
                "title": "street géométrique",
                "url_s": "https://live.staticflickr.com/7866/47541388921_213f95c299_m.jpg",
                "width_s": "160",
              }
            }
          />
          <Feed
            data={
              Object {
                "description": Object {
                  "_content": "OLYMPUS DIGITAL CAMERA",
                },
                "farm": 8,
                "height_s": "156",
                "id": "46817753384",
                "isfamily": 0,
                "isfriend": 0,
                "ispublic": 1,
                "owner": "163510631@N05",
                "ownername": "evablanchardcouet",
                "secret": "3a8d43ea24",
                "server": "7926",
                "tags": "nature birds tree outside",
                "title": "birds , birds , birds",
                "url_s": "https://live.staticflickr.com/7926/46817753384_3a8d43ea24_m.jpg",
                "width_s": "240",
              }
            }
          />
        </MasonryComponent>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Feed
              data={
                Object {
                  "description": Object {
                    "_content": "Podiceps cristatus
Great Crested Grebe
Peschiera del Garda",
                  },
                  "farm": 8,
                  "height_s": "160",
                  "id": "40575384753",
                  "isfamily": 0,
                  "isfriend": 0,
                  "ispublic": 1,
                  "owner": "59464817@N02",
                  "ownername": "Fausto Deseri",
                  "secret": "310af33489",
                  "server": "7828",
                  "tags": "greatcrestedgrebe podicepscristatus svassomaggiore corteggiamento peschieradelgarda wildlife nature birds wildanimals nikond500 afsnikkor300mmf28difedii nikontc17eii faustodeseri",
                  "title": "Svassi Maggiori",
                  "url_s": "https://live.staticflickr.com/7828/40575384753_310af33489_m.jpg",
                  "width_s": "240",
                }
              }
            />,
            <Feed
              data={
                Object {
                  "description": Object {
                    "_content": "",
                  },
                  "farm": 8,
                  "height_s": "240",
                  "id": "47541388921",
                  "isfamily": 0,
                  "isfriend": 0,
                  "ispublic": 1,
                  "owner": "115045266@N08",
                  "ownername": "Rudy Pilarski",
                  "secret": "213f95c299",
                  "server": "7866",
                  "tags": "structure street nikon nb bw bâtiment building blackandwhite noiretblanc monochrome perspective paris people personne geometry géométrie géométria géométrique graphique graphic urbain urban urbano urbanisme france francia europe europa jussieux architecture architectura architectural dowtown thebestoffnikon thepassionphotography d7100 design",
                  "title": "street géométrique",
                  "url_s": "https://live.staticflickr.com/7866/47541388921_213f95c299_m.jpg",
                  "width_s": "160",
                }
              }
            />,
            <Feed
              data={
                Object {
                  "description": Object {
                    "_content": "OLYMPUS DIGITAL CAMERA",
                  },
                  "farm": 8,
                  "height_s": "156",
                  "id": "46817753384",
                  "isfamily": 0,
                  "isfriend": 0,
                  "ispublic": 1,
                  "owner": "163510631@N05",
                  "ownername": "evablanchardcouet",
                  "secret": "3a8d43ea24",
                  "server": "7926",
                  "tags": "nature birds tree outside",
                  "title": "birds , birds , birds",
                  "url_s": "https://live.staticflickr.com/7926/46817753384_3a8d43ea24_m.jpg",
                  "width_s": "240",
                }
              }
            />,
          ],
          "className": "",
          "disableImagesLoaded": false,
          "elementType": "div",
          "enableResizableChildren": false,
          "imagesLoadedOptions": Object {},
          "onLayoutComplete": [Function],
          "onRemoveComplete": [Function],
          "options": Object {},
          "updateOnEachImageLoad": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "description": Object {
                  "_content": "Podiceps cristatus
Great Crested Grebe
Peschiera del Garda",
                },
                "farm": 8,
                "height_s": "160",
                "id": "40575384753",
                "isfamily": 0,
                "isfriend": 0,
                "ispublic": 1,
                "owner": "59464817@N02",
                "ownername": "Fausto Deseri",
                "secret": "310af33489",
                "server": "7828",
                "tags": "greatcrestedgrebe podicepscristatus svassomaggiore corteggiamento peschieradelgarda wildlife nature birds wildanimals nikond500 afsnikkor300mmf28difedii nikontc17eii faustodeseri",
                "title": "Svassi Maggiori",
                "url_s": "https://live.staticflickr.com/7828/40575384753_310af33489_m.jpg",
                "width_s": "240",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "description": Object {
                  "_content": "",
                },
                "farm": 8,
                "height_s": "240",
                "id": "47541388921",
                "isfamily": 0,
                "isfriend": 0,
                "ispublic": 1,
                "owner": "115045266@N08",
                "ownername": "Rudy Pilarski",
                "secret": "213f95c299",
                "server": "7866",
                "tags": "structure street nikon nb bw bâtiment building blackandwhite noiretblanc monochrome perspective paris people personne geometry géométrie géométria géométrique graphique graphic urbain urban urbano urbanisme france francia europe europa jussieux architecture architectura architectural dowtown thebestoffnikon thepassionphotography d7100 design",
                "title": "street géométrique",
                "url_s": "https://live.staticflickr.com/7866/47541388921_213f95c299_m.jpg",
                "width_s": "160",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "description": Object {
                  "_content": "OLYMPUS DIGITAL CAMERA",
                },
                "farm": 8,
                "height_s": "156",
                "id": "46817753384",
                "isfamily": 0,
                "isfriend": 0,
                "ispublic": 1,
                "owner": "163510631@N05",
                "ownername": "evablanchardcouet",
                "secret": "3a8d43ea24",
                "server": "7926",
                "tags": "nature birds tree outside",
                "title": "birds , birds , birds",
                "url_s": "https://live.staticflickr.com/7926/46817753384_3a8d43ea24_m.jpg",
                "width_s": "240",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
